packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -51,7 +56,7 @@ class DBot {
             // Push main.xml to workspace and reset the undo stack.
             Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(main_xml), this.workspace);
             this.workspace.clearUndo();
-
+            this.workspace.cleanUp();
```

Comment:
I think it's better to put this before `clearUndo` as `cleanUp` will add an event (or multiple) to the undo/redo stack.

Reply:
Done!